packages/hoppscotch-common/src/helpers/import-export/import/openapi.ts

```diff
@@ -54,6 +56,22 @@ const objectHasProperty = <T extends string>(
   typeof obj === "object" &&
   Object.prototype.hasOwnProperty.call(obj, propName)
 
+// Helper function to check for unresolved references in a document
+const hasUnresolvedRefs = (obj: any): boolean => {
+  if (!obj || typeof obj !== "object") return false
+
+  // Check if current object has $ref property
+  if (obj.$ref && typeof obj.$ref === "string") return true
+
+  // Check arrays
+  if (Array.isArray(obj)) {
+    return obj.some((item) => hasUnresolvedRefs(item))
+  }
+
+  // Check object properties
+  return Object.values(obj).some((value) => hasUnresolvedRefs(value))
+}
```

Comment:
This will fail for incoming objects with circular references. For instance, attempting to import the OpenAPI spec [here](https://github.com/hoppscotch/hoppscotch/issues/3554#issuecomment-1898009615) will result in a stack overflow exception.